official/resnet/keras/keras_imagenet_test.py

```diff
@@ -0,0 +1,491 @@
+    if context.num_gpus() < 2:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(2, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "2",
+        "-dtype", "fp16",
+        "-enable_eager", "true",
+        "-enable_xla", "true",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_imagenet_xla_2_gpu_fp16_tensorboard_tweaked",
+        "-tf_gpu_thread_mode", "gpu_private",
+        "-enable_tensorboard", "true",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_imagenet_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_2_gpu(self):
+    """Test Keras model in legacy graph mode with 2 GPUs."""
+    if context.num_gpus() < 2:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(2, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "2",
+        "-enable_eager", "false",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_imagenet_graph_2_gpu",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_imagenet_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_xla_2_gpu(self):
+    """Test Keras model in legacy graph mode with XLA and 2 GPUs."""
+    if context.num_gpus() < 2:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(2, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "2",
+        "-enable_eager", "false",
+        "-enable_xla", "true",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_imagenet_graph_xla_2_gpu",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_imagenet_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
+  def test_end_to_end_graph_xla_2_gpu_fp16_tweaked(self):
+    """Test Keras model in legacy graph mode with manual config tuning, XLA,
+       2 GPUs and fp16.
+    """
+    if context.num_gpus() < 2:
+      self.skipTest(
+          "{} GPUs are not available for this test. {} GPUs are available".
+          format(2, context.num_gpus()))
+
+    extra_flags = [
+        "-num_gpus", "2",
+        "-dtype", "fp16",
+        "-enable_eager", "false",
+        "-enable_xla", "true",
+        "-distribution_strategy", "default",
+        "-model_dir", "keras_imagenet_graph_xla_2_gpu_fp16_tweaked",
+        "-tf_gpu_thread_mode", "gpu_private",
+    ]
+    extra_flags = extra_flags + self._extra_flags
+
+    integration.run_synthetic(
+        main=keras_imagenet_main.run,
+        tmp_root=self.get_temp_dir(),
+        extra_flags=extra_flags
+    )
+
```

Comment:
yeah we should definitely have much lesser cases in this test suite

Reply:
Reduced the cases. There are 10 cases now, which I think is reasonable.